Library/src/lab1/Library.java

```diff
@@ -2,10 +2,82 @@
 
 /**
  * A representation of a library of books.
- * @author <a href="mailto:rkhatchadourian@citytech.cuny.edu">Raffi Khatchadourian</a>.
+ *
+ * @author jianlang lin,Ziying Guo
  */
 public class Library {
-    // TODO: Add the missing implementation to this class
+
+    String library_address;
+    static String library_hours = "Libraries are open daily from 9AM to 5PM";
+    Book[] catalog = new Book[0];
+    int numberofBooks = 0;
+
+//List of Methods for Libraries
+    public Library(String address) {
+        library_address = address;
+    }
+
+    public static void printOpeningHours() {
+        System.out.println(library_hours);
+    }
+
+    public void printAddress() {
+        System.out.println(library_address);
+    }
+
+    //Add books to library catalog
+    public void addBook(Book newbook) {
+        if (numberofBooks >= 0) {
+            Book[] newarray = new Book[(numberofBooks + 1)];
```

Comment:
I think I would have a partially filled array here instead of creating a new one each time.